lib/TAP/Formatter/Session/TeamCity.pm

```diff
@@ -454,34 +451,59 @@ sub _finish_suite {
 
     return 0 unless $name eq $self->_tc_suite_name_stack->[-1];
 
-    if ( $self->_tc_is_last_suite_empty ) {
-        my $suite_type
-            = @{ $self->_tc_suite_name_stack } == 1 ? 'file' : 'subtest';
-        my $test_name   = "Test died before reaching end of $suite_type";
-        my $test_result = TAP::Parser::Result::Test->new(
-            {
-                'ok'          => 'not ok',
-                'explanation' => q{},
-                'directive'   => q{},
-                'type'        => 'test',
-                'test_num'    => 1,
-                'description' => "- $test_name",
-                'raw'         => "not ok 1 - $test_name",
-            }
-        );
-        $self->_test_started($test_result);
+    if ( $self->_tc_last_suite_is_empty ) {
+        $self->_test_started( $self->_test_died_result );
         $self->_tc_test_output_buffer( $self->_tc_suite_output_buffer );
         $self->_tc_suite_output_buffer(q{});
         $self->_test_finished;
     }
     pop @{ $self->_tc_suite_name_stack };
     $self->_tc_suite_output_buffer(q{});
-    $self->_tc_is_last_suite_empty(0);
+    $self->_tc_last_suite_is_empty(0);
     $self->_tc_message( 'testSuiteFinished', { name => $name } );
 
     return 1;
 }
 
+sub _append_to_tc_test_output_buffer {
+    my $self   = shift;
+    my $output = shift;
+
+    $self->_tc_test_output_buffer( $self->_tc_test_output_buffer . $output );
+
+    return;
+}
+
+sub _append_to_tc_suite_output_buffer {
+    my $self   = shift;
+    my $output = shift;
+
+    $self->_tc_suite_output_buffer(
+        $self->_tc_suite_output_buffer . $output );
+
+    return;
+}
+
+sub _test_died_result {
+
+    # We used to try to figure out whether we died in a subtest or the top
```

Comment:
This comment isn't very clear.  What is it that `_test_died_result` actually does? (the commit message is a _little_ clearer, but that's not in the code)